Geometry/MTDNumberingBuilder/plugins/CmsMTDConstruction.cc

```diff
@@ -80,3 +68,23 @@ void CmsMTDConstruction::buildETLModule(DDFilteredView& fv, GeometricTimingDet*
 
   mother->addComponent(det);
 }
+
+GeometricTimingDet* CmsMTDConstruction::buildSubdet(DDFilteredView& fv,
+                                                    GeometricTimingDet* mother,
+                                                    const std::string& attribute) {
+  auto thisDet = theCmsMTDStringToEnum.type(fv.name());
+  GeometricTimingDet* subdet = new GeometricTimingDet(&fv, thisDet);
```

Comment:
A better pattern would be for the mother volume to create the subdetector as a smart pointer that it already owns, and then return it to this code for configuration. This would avoid a bare `new `, which always makes me wary of possible memory leaks. This could be addressed in a followup PR, since I think this is a common issue throughout the geometry code.

Reply:
it is a common pattern in the parallel tracker code, from which this code is derived, and this PR reduces the number of instances. I would prefer to defer this update to a later moment, as there are already too many things boiling in the plate at present (new ETL, DD4hep migration)